ui/components/app/auto-detect-token/auto-detect-token-modal.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { useCallback, useContext } from 'react';
+
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  Modal,
+  ModalContent,
+  ModalOverlay,
+  ModalHeader,
+  Box,
+  Text,
+  ModalBody,
+  ModalFooter,
+} from '../../component-library';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import {
+  AlignItems,
+  BorderRadius,
+  Display,
+  FlexDirection,
+  JustifyContent,
+  TextAlign,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+import { setUseTokenDetection } from '../../../store/actions';
+import { MetaMetricsContext } from '../../../contexts/metametrics';
+import {
+  MetaMetricsEventCategory,
+  MetaMetricsEventName,
+} from '../../../../shared/constants/metametrics';
+import { getProviderConfig } from '../../../ducks/metamask/metamask';
+import { getCurrentLocale } from '../../../ducks/locale/locale';
+import { ORIGIN_METAMASK } from '../../../../shared/constants/app';
+
+type AutoDetectTokenModalProps = {
+  isOpen: boolean;
+  onClose: (arg: boolean) => void;
+  setShowTokenAutodetectModalOnUpgrade: (arg: boolean) => void;
+};
+function AutoDetectTokenModal({
+  isOpen,
+  onClose,
+  setShowTokenAutodetectModalOnUpgrade,
+}: AutoDetectTokenModalProps) {
+  const t = useI18nContext();
+  const dispatch = useDispatch();
+  const trackEvent = useContext(MetaMetricsContext);
+  const { chainId } = useSelector(getProviderConfig);
+
+  const handleTokenAutoDetection = useCallback(
+    (val) => {
+      trackEvent({
+        event: val
+          ? MetaMetricsEventName.TokenAutoDetectionEnableModal
+          : MetaMetricsEventName.TokenAutoDetectionDisableModal,
+        category: MetaMetricsEventCategory.Navigation,
+        properties: {
+          chain_id: chainId,
+          locale: getCurrentLocale,
+          referrer: ORIGIN_METAMASK,
+        },
+      });
+      dispatch(setUseTokenDetection(val));
+      onClose(val);
+      setShowTokenAutodetectModalOnUpgrade(val);
+    },
+    [dispatch],
+  );
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      onClose={() => onClose(true)}
+      isClosedOnOutsideClick={false}
+      isClosedOnEscapeKey={false}
+      className="mm-modal__custom-scrollbar auto-detect-in-modal"
+      autoFocus={false}
+    >
+      <ModalOverlay />
+      <ModalContent
+        modalDialogProps={{ className: 'auto-detect-token-modal__dialog' }}
```

Comment:
Is this className needed? I saw the css file has `auto-detect-in-modal ` instead of `auto-detect-token-modal`